pytorch3dunet/augment/transforms.py

```diff
@@ -549,14 +549,26 @@ class Normalize:
     Apply simple min-max scaling to a given input tensor, i.e. shrinks the range of the data in a fixed range of [-1, 1].
     """
 
-    def __init__(self, min_value, max_value, **kwargs):
-        assert max_value > min_value
+    def __init__(self, min_value=None, max_value=None, norm01=False, **kwargs):
+        if min_value is not None and max_value is not None:
+          assert max_value > min_value
         self.min_value = min_value
-        self.value_range = max_value - min_value
+        self.max_value = max_value
+        self.norm01 = norm01
 
     def __call__(self, m):
+        if self.min_value is None:
+          self.min_value = np.min(m)
```

Comment:
This implementation has errors:
1. It changes the min/max values in the 1st invocation, so subsequent invocation of this transformer will have incorrect min/max values
2. It makes it not thread-safe which will fail with multi-threaded data loading scenario

Reply:
Thanks for your review. Please see if my latest changes fixes the issues and is enough to be merged.